.github/workflows/docker-deploy.yml

```diff
@@ -1,29 +1,67 @@
-name: Docker Deploy
+name: Build and Publish Docker Images to GHCR.io
 
 on:
   push:
-    branches:
-      - main
```

Comment:
I removed this since this will publish an image on each branch/PR with the branch name as a test which is helpful for testing.

`latest` will be tagged only on the default branch (so after merge)